project-koku/koku
# Fetching Sources imported field for OCP cluster ID if available.

Current implementation is to use the Source guid as the OCP cluster ID in cost management.  This change allows for a user supplied cluster-id to be associated with an OCP Platform Source.

This ID would be returned from [Korekuta](https://github.com/project-koku/korekuta) and entered by the user on the Platform Sources UI when associated Cost Management to the OpenShift source.

I kept the Source guid fallback in place so that any attempts to create an OCP Cost Source in CI would be "successful".  Once the front end implements the OCP Cost Management pages we could remove the fallback.

@agrare we discussed this briefly but I wanted to get your sign off before proceeding.  Here's an example of what the sources table would look like for an OCP source with and without a provided cluster id.  Do you want this `imported` field to be a more standard json structure to be more descriptive or is the simple UUID ok? 

```
sources_api_development=# select * from sources;
 id |            name            |                 uid                  |         created_at         |         updated_at         | tenant_id | source_type_id | version | availability_status |               imported
----+----------------------------+--------------------------------------+----------------------------+----------------------------+-----------+----------------+---------+---------------------+--------------------------------------
  1 | Test OCP Source cluster-id | b6255061-4330-4a5e-9214-62ebffc5dde1 | 2019-10-28 20:53:48.908224 | 2019-10-28 20:53:48.908224 |         1 |              1 |         |                     | 8770d031-9b46-40d9-be12-52929af682f5
  2 | Test OCP Source            | 3adbef49-e9f3-4cb9-bee3-3d834c837bd3 | 2019-10-28 20:54:58.993587 | 2019-10-28 20:54:58.993587 |         1 |              1 |         |                     |
(2 rows)
```


**Testing**
1. Create OCP Source with cluster-id provided.
2. Create OCP source without cluster-id provided.  Verify it falls back to Source uid.
3. Verify that the Sources DB has entries as expected

**Test Results**
[ocp_sources_cluster_id_ut.txt](https://github.com/project-koku/koku/files/3795775/ocp_sources_cluster_id_ut.txt)

Addresses #1310
